lib/ansible/modules/cloud/vmware/vmware_local_user_facts.py

```diff
@@ -50,56 +52,108 @@
     type: dict
     sample: [
         {
-            "full_name": "Administrator",
```

Comment:
We are changing facts keys here, we should add older values as well as new values. Because, removing old values will break playbooks and roles which are already written by user of this modules.
Ideally we should 

1. Add a note in notes section stating these values will be deprecated after 4 versions.
2. Add a [porting guide](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guides/porting_guide_2.8.rst)

Let me know if it makes sense.

Reply:
Yes, I understand it is a bit overkill, but consider from UX point of view. People already using this module will be surprised to see failing PB which were already working. 

You can keep - 
```
        {
            "full_name": "DCUI User",
            "principal": "dcui",
            "user_group": false,
            "role": "admin",
            "description": "DCUI User",
            "group": false,
            "user_id": 100,
            "user_shell_access": false
            "user_name": "dcui",
            "shell_access": false
        },
```

I aware that information is redundant but that will atleast not break playbooks. We will add porting guide note [here](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guides/porting_guide_2.8.rst#noteworthy-module-changes) and a note in documentation stating that these values will be deprecated after 4 release i.e. `2.12`

Reply:
I have added similar note in porting guide for `vmware_local_role_facts` here = https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guides/porting_guide_2.8.rst#id12

Reply:
ok. I thought this is the PR with the dict to list change. I added the old names and notes.

Reply:
What do you want to write in the port guide? I think it's sufficient if we add a note later in 2.12 that the properties are deprecated.